plugin/pkg/dnstest/server.go

```diff
@@ -23,18 +23,28 @@ func NewServer(f dns.HandlerFunc) *Server {
 	ch1 := make(chan bool)
 	ch2 := make(chan bool)
 
-	l, _ := net.Listen("tcp", ":0")
-	if l == nil {
-		return nil
+	s1 := &dns.Server{} // udp
+	s2 := &dns.Server{} // tcp
+
+	for i := 0; i < 1000; i++ { // 1000 attempts
+		s2.Listener, _ = net.Listen("tcp", ":0")
+		if s2.Listener == nil {
+			panic("dnstest.NewServer(): failed to create tcp listener")
```

Comment:
shouldn't this be `continue`?

Reply:
Here we ask system for **any** available port, so if it fails here this means that system has no available ports or some error not related to port. So, I don't think we should retry in this case